cloud-reactor-api/src/main/java/com/sequenceiq/cloudbreak/cloud/service/ResourceRetrieval.java

```diff
@@ -0,0 +1,12 @@
+package com.sequenceiq.cloudbreak.cloud.service;
+
+import java.util.Optional;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.common.api.type.CommonStatus;
+import com.sequenceiq.common.api.type.ResourceType;
+
+public interface ResourceRetrieval {
```

Comment:
Also, there are 3 interfaces for some kind of manipulation. It might make sense to apply a naming structure, like ResourcePersister - ResourceNamer - ResourceRetriever. Or, add a new package named as resource and move the classes there as Persister - Namer - Retriever. This could go as a followup refactoring jira.